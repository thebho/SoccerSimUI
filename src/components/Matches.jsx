// @flow
import React , { Component } from 'react';
import type { Match } from '../model';
import {
  Table,
  TableBody,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';

const styles = {
  tableCol: {
    height: '25px',
    textAlign: 'right',
    fontSize: '12px',
  }
}

type Props = {
  weeklyMatches?: Array<Match>
}

class Matches extends Component<Props> {
  render() {
  return (
      <Table
      adjustForCheckbox={false}
      className="Table"
      displaySelectAll={false}
      >
      <TableBody
        displayRowCheckbox={false}
      >
        {this.props.weeklyMatches ? this.props.weeklyMatches.map(match => (
          <TableRow
            style={{height: '25px'}}
            key={match.ID}
          >
            <TableRowColumn style={styles.tableCol}>
              {match.HomeTeam}  {match.HomeTeamGoals}
            </TableRowColumn>
            <TableRowColumn style={styles.tableCol}>
              {match.AwayTeam}  {match.AwayTeamGoals}
            </TableRowColumn>
          </TableRow>
          )): null}
        </TableBody>
      </Table>
  )}
}

export default Matches;
