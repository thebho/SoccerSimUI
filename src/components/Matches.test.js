import React from 'react';
import ReactDOM from 'react-dom';
import { fn } from 'jest';
import Matches from './Matches';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

const mockMatches = [
  {
    ID: 1,
    HomeTeam: 'ARS',
    AwayTeam: 'TOT',
    HomeTeamGoals: null,
    AwayTeamGoals: null,
    MatchWeek: 1,
    Season: 'TestSeason',
  },
  {
    ID: 2,
    HomeTeam: 'WHA',
    AwayTeam: 'CHE',
    HomeTeamGoals: null,
    AwayTeamGoals: null,
    MatchWeek: 1,
    Season: 'TestSeason',
  },
];

describe('TeamsTable', () => {
  it('matches snapshot with matches', () => {
    const wrapper = shallow(
      <MuiThemeProvider>
        <Matches weeklyMatches={mockMatches} />
      </MuiThemeProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
