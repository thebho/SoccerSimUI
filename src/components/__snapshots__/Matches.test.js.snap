// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsTable matches snapshot with matches 1`] = `
<Matches
  weeklyMatches={
    Array [
      Object {
        "AwayTeam": "TOT",
        "AwayTeamGoals": null,
        "HomeTeam": "ARS",
        "HomeTeamGoals": null,
        "ID": 1,
        "MatchWeek": 1,
        "Season": "TestSeason",
      },
      Object {
        "AwayTeam": "CHE",
        "AwayTeamGoals": null,
        "HomeTeam": "WHA",
        "HomeTeamGoals": null,
        "ID": 2,
        "MatchWeek": 1,
        "Season": "TestSeason",
      },
    ]
  }
/>
`;
